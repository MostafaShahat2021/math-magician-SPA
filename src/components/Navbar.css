nav {
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

nav a {
  text-decoration: none;
  color: black;
  border-right: 1px solid black;
  padding-right: 1rem;
  margin: 1rem;
}

nav a.active {
  font-weight: bold;
  color: rgb(81, 178, 182);
  border-right: 1px solid black;
  text-decoration: underline;
}