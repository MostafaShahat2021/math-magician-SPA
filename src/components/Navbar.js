import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navbar.css';

function Navbar() {
  return (
    <div className="calc-cont">
      <h1>Math Magicians</h1>
      <nav>
        <NavLink to="/">Home</NavLink>
        <NavLink to="/calculator">Calculator</NavLink>
        <NavLink to="quote">Quote</NavLink>
      </nav>
    </div>
  );
}

export default Navbar;
