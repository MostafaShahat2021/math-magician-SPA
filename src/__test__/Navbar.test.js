import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

it('Navbar should render', () => {
  const navElement = render(<BrowserRouter><Navbar /></BrowserRouter>);
  expect(navElement).toMatchSnapshot();
});
