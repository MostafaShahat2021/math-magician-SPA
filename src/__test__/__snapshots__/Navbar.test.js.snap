// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calc-cont"
      >
        <h1>
          Math Magicians
        </h1>
        <nav>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
          <a
            class=""
            href="/calculator"
          >
            Calculator
          </a>
          <a
            class=""
            href="/quote"
          >
            Quote
          </a>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calc-cont"
    >
      <h1>
        Math Magicians
      </h1>
      <nav>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          class=""
          href="/calculator"
        >
          Calculator
        </a>
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
